sdl_android/src/main/java/com/smartdevicelink/api/lockscreen/LockScreenManager.java

```diff
@@ -0,0 +1,274 @@
+	 * usually call launchLockScreenActivity
+	 *
+	 * 1. ON_HMI_STATUS
+	 * 2. ON_DRIVER_DISTRACTION
+	 * 3. ON_SYSTEM_REQUEST (used for OEM Icon Downloading)
+	 */
+	private void setupListeners(){
+		// add hmi listener
+		hmiListener = new OnRPCNotificationListener() {
+			@Override
+			public void onNotified(RPCNotification notification) {
+				hmiLevel = ((OnHMIStatus)notification).getHmiLevel();
+				launchLockScreenActivity();
+			}
+		};
+		internalInterface.addOnRPCNotificationListener(FunctionID.ON_HMI_STATUS, hmiListener);
+
+		// set up driver distraction listener
+		ddListener = new OnRPCNotificationListener() {
+			@Override
+			public void onNotified(RPCNotification notification) {
+				// do something with the status
+				if (notification != null) {
+					OnDriverDistraction ddState = (OnDriverDistraction) notification;
+
+					if (ddState.getState() == DriverDistractionState.DD_ON){
+						Log.i(TAG, "DD On Notification");
+						// launch lock screen
+						driverDistStatus = true;
+						launchLockScreenActivity();
+					}else{
+						Log.i(TAG, "DD Off Notification");
+						// close lock screen
+						driverDistStatus = false;
+						context.get().sendBroadcast(new Intent(SDLLockScreenActivity.CLOSE_LOCK_SCREEN_ACTION));
+					}
+				}
+			}
+		};
+		internalInterface.addOnRPCNotificationListener(FunctionID.ON_DRIVER_DISTRACTION, ddListener);
+
+		// set up system request listener
+		if (showOEMLogo) {
+			systemRequestListener = new OnRPCNotificationListener() {
+				@Override
+				public void onNotified(RPCNotification notification) {
+					// do something with the status
+					final OnSystemRequest msg = (OnSystemRequest) notification;
+					try {
+						Log.i(TAG, "SYSTEM REQUEST: " + notification.serializeJSON().toString());
+					} catch (JSONException e) {
+						e.printStackTrace();
+					}
+					if (msg.getRequestType() == RequestType.LOCK_SCREEN_ICON_URL &&
+							msg.getUrl() != null) {
+						// send intent to activity to download icon from core
+						Log.i(TAG, "SYSTEM REQUEST - ICON Ready for Download");
+						OEMIconUrl = msg.getUrl();
+						downloadLockScreenIcon(OEMIconUrl);
+					}
+				}
+			};
+			internalInterface.addOnRPCNotificationListener(FunctionID.ON_SYSTEM_REQUEST, systemRequestListener);
+		}
+	}
+
+	////
+	// LAUNCH LOCK SCREEN LOGIC
+	////
+
+	/**
+	 * 1. Check if user wants us to manage lock screen
+	 * 2. If so, get the HMI level and LockScreenStatus from the method below
+	 * 3. Build intent and start the SDLLockScreenActivity
+	 *
+	 * X. If the status is set to OFF, Send broadcast to close lock screen if it is open
+	 */
+	private void launchLockScreenActivity(){
+		// intent to open SDLLockScreenActivity
+		// pass in icon, background color, and custom view
+		if (lockScreenEnabled) {
+			LockScreenStatus status = getLockScreenStatus();
+			if (hmiLevel == HMILevel.HMI_FULL && status == LockScreenStatus.REQUIRED) {
```

Comment:
Checking for HMI_FULL here seems redundant. The `getLockScreenStatus` call already checks for hmi level, am I missing something?